### DISCUSSION TITLE
This document uses unsuported features (XFA) and might not render properly

### DISCUSSION THREAD

**Discussion by snowbound** [sumatrapdfreader/sumatrapdf#2872] [General]
Using SumatraPDF 3.4.6 on Win10 21H2 and all of a sudden Sumatra is showing for every PDF file I have tried that "This document uses unsuported features (XFA) and might not render properly". Was this something that Sumatra decided not to support? Only reference I found was a post back in 2016.

**Comment by GitHubRulesOK** [comment]
The error/warning message can "stick" until file is dismissed and XFA files are a proprietary Adobe Dynamic format. That .pdf extension should not have been used but reserved for standard.PDF screen/printer files, but those were the days when Acrobat ruled.

**Reply by snowbound** [reply]
The thing is that I have been using SumatraPDF reader for years and until this week have never seen this warning. I just tried a 5 year old PDF file that Sumatra would have opened up in the past. All of my PDFs are exhibiting this warning including manuals that were downloaded from the manufacturer's websites over the years.

**Reply by GitHubRulesOK** [reply]
definitely something amiss I rarely see that warning unless a file is stuck in history to trigger on session reload
running a copy of 3.4.6 I cant get that message until I open an xfa file and then each time the session attempts to load that file its repeated then close that tab the message will NOT come back on next file open.

The big issue is knowing which file tab is triggering the warning.

**Reply by snowbound** [reply]
That is really weird. As I said this warning is something I never saw before the last couple of days and was wondering if it was a change made to SumatraPDF code. I have multiple tabs open but now I can click on any PDF file (whose icon in Windows FIle explorer is another PDF reader I have installed rather than Sumatra yellow icon) and it will generate that warning. I close that tab and select another PDF same warning and I can repeat the process over and over again.

It seems that this warning is being propagated across all open tabs and shows up on the Frequently Read page if you close all of the other tabs. If the Frequently Read page is the only page open and shows the warning then restarting Sumatra causes the warning to disappear.  If you happen to open a PDF that triggers the warning then all subsequent file openings will also trigger that warning to show up in all the other tabs.

**Reply by snowbound** [reply]
This is really weird. I closed all the tabs in Sumatra until I got only the Frequently Read page. Even that had the warning showing up. I then closed Sumatra and restarted it and clicked on any of the documents on the Frequently Read page and none of the documents raised the warning. This is really weird since I did not suspend the warning on any of these documents in the past so if the XFA warning should of reappeared if it was a valid warning. It appears Sumatra was generating false warnings for some reason.

**Reply by GitHubRulesOK** [reply]
***that's what I meant by "sticky" once triggered it is not dismissed unless the tab that triggers the warning is dismissed***  

![image](https://user-images.githubusercontent.com/12754913/185028625-d885442e-48dd-4d39-a741-80d865f4070d.png)
![image](https://user-images.githubusercontent.com/12754913/185028992-ed920e7e-9620-49d4-b10f-4d174d9a213a.png)

**Reply by snowbound** [reply]
In my case, I had no idea which of a dozen or more tabs was causing the issue since each of them was being displayed properly and it was not an overlaid warning like in your example. The warning was all by itself in the upper left corner of each of the tabs with a X in a box. Only discovered that it was this particular window when I closed all the tabs and loaded them up 1 by 1.

![image](https://user-images.githubusercontent.com/596147/185159659-48c2e730-c3a4-414e-b9ec-1bba55476c2d.png)

**Reply by GitHubRulesOK** [reply]
Hmm that would do it 
but best practice for authors using XFA is it "should" include a blocking cover page (as in my example) to show it is not standard pdf, MS use something similar for their custom type pdf files. perhaps open an issue to include tab name in warning

Official line is its not possible to support XFA.pdf unless MuPDF has such ability, see background info in https://github.com/sumatrapdfreader/sumatrapdf/issues/48#issuecomment-406830166
